### ISSUE TITLE
Binary ConfigMap referenced by workload - Javascript Error in Rancher UI 

### ISSUE THREAD

**Issue by kengibous** [rancher/rancher#21349]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**

Create a configMap with binary data - I am using this configMap to write a certificate to a pod's filesystem

Example -
```
apiVersion: v1
binaryData:
  lower.data.protection.pfx: <base64encode cert data here>
kind: ConfigMap
```

Then create a workload that references the configMap. 

Example -
```
apiVersion: apps/v1beta2
kind: Deployment
metadata:
  ...
      volumes:
      - configMap:
          defaultMode: 420
          items:
          - key: lower.data.protection.pfx
            path: dp.pfx
          name: lowerdataprotectioncert-config
        name: dataprotection
```

Then in the rancher user interface if you click the workload, a javascript error occurs and the page is stuck with a loading icon.

**Result:**

![image](https://user-images.githubusercontent.com/6308070/60731497-4b67dd00-9f0d-11e9-94b2-499483abc6d7.png)

Java script error - 

component.js:144 Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at n.updateConfigMapKeys (component.js:144)
    at n.init (component.js:44)
    at n [as init] (0:58190)
    at n.e (0:57415)
    at n (0:57475)
    at n (0:57475)
    at n (0:57475)
    at new n (0:57475)
    at Function.e.create (0:57488)

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.2
- Installation option (single install/HA): single install

**Comment by westlywright** [comment]
Available in `latest2`.

**Comment by izaac** [comment]
Rancher version: `2.4 master-head (11/22/2019) 2a08c3a9f`

Using `dataBinary` in ConfigMap is no longer makes the UI to crash or show any warnings.

However the key/value pair won't show up in the ConfigMap Details in the UI which kind of makes sense as it's probably a bad idea to try render binary data.

![Screen Shot 2019-11-22 at 3 17 49 PM](https://user-images.githubusercontent.com/27659/69464580-e84fd900-0d3b-11ea-810a-4929a3b9da52.png)


There's no key to select during the workload creation in the Rancher UI. I just had to use `All Keys`

![Screen Shot 2019-11-22 at 3 19 25 PM](https://user-images.githubusercontent.com/27659/69464602-f69df500-0d3b-11ea-95d0-1ed7ebda05ff.png)

Making this and clicking in the workload I was able to see the details and even open the shell and check binary file mounted in the path specified.

![Screen Shot 2019-11-22 at 3 18 13 PM](https://user-images.githubusercontent.com/27659/69464623-04537a80-0d3c-11ea-8676-c8b0e787925b.png)


@sangeethah @deniseschannon UI is no longer crashing but the key won't show in the UI as it's data is binary if we navigate to the ConfigMap details in Rancher. There's also no way to specify the key when configuring the volume in the workload. I had to use `All keys`.
